src/app/lib/views/browser/item.js

```diff
@@ -77,247 +71,216 @@
-                    .then(function () {
-                        that.model.set('watched', false);
-                        App.vent.trigger('movie:unwatched', {
-                            imdb_id: that.model.get('imdb_id')
-                        }, 'seen');
+                this.$el.removeClass('watched');
+
+                // set cover state
+                this.setCoverStates();
+                this.setTooltips();
+
+                // event propagation
+                App.vent.trigger(appEvent, {
+                    imdb_id: imdb
+                }, 'seen');
+            }.bind(this));
+        },
+
+        addBookmarked: function () {
+            var imdb = this.model.get('imdb_id');
+            var itemtype = this.model.get('type');
+            var provider = App.Providers.get(this.model.get('provider'));
+
+            return provider.detail(imdb, this.model.attributes).then(function (data) {
+                data.provider = provider.name;
+
+                if (itemtype === 'show') {
+                    promisifyDb(db.shows.find({
+                        imdb_id: imdb.toString(),
+                    })).then(function (res) {
+                        if (res != null && res.length > 0) {
+                            return Database.updateTVShow(data);
+                        } else {
+                            return Database.addTVShow(data);
+                        }
                     });
-            } else {
-                this.ui.watchedIcon.addClass('selected');
-                switch (Settings.watchedCovers) {
-                case 'fade':
-                    this.$el.addClass('watched');
-                    break;
-                case 'hide':
-                    this.$el.remove();
-                    break;
+                } else {
+                    return Database.addMovie(data);
                 }
-                Database.markMovieAsWatched({
-                        imdb_id: this.model.get('imdb_id'),
-                        from_browser: true
-                    }, true)
-                    .then(function () {
-                        that.model.set('watched', true);
-                        App.vent.trigger('movie:watched', {
-                            imdb_id: that.model.get('imdb_id')
-                        }, 'seen');
-                    });
+            }).then(function () {
+                return Database.addBookmark(imdb, itemtype);
+            }).then(function () {
+                win.info('Bookmark added (' + imdb + ')');
+            }.bind(this));
+        },
 
-            }
+        removeBookmarked: function () {
+            var imdb = this.model.get('imdb_id');
+            var itemtype = this.model.get('type');
+            var dbCall = (function () {
+                if (itemtype.match('show')) {
+                    return 'deleteTVShow';
+                } else {
+                    return 'deleteMovie';
+                }
+            }());
```

Comment:
because it was movies only. Here we can add 'short' or w/e future thing we implement.

Above : watched: Boolean
Here: itemtype: String